dali/python/nvidia/dali/pipeline.py

```diff
@@ -209,6 +209,7 @@ def __init__(self,
         self._py_pool_started = False
         self._backend_prepared = False
         self._built = False
+        self._deserialized = False  # Marked True when deserializing
```

Comment:
This is too much of an assumption. There can be a legitimate sourceless external source that is not deserialized (this is actually used in tests). Something like `has_source` or simply checking the original `source` would be more adequate.

Reply:
I don't follow? If we don't deserialize, the check works as intended, as we build the pipeline via frontend, and all the members there are valid. And this is just a flag that signifies that this pipeline was deserialized, I added one, instead of relying on checking the `not self._py_graph_built and self._backend_prepared and self_built` to check if we are in deserialized pipeline - as that might not be enough.